frontend/src/views/RouteFormView.vue

```diff
@@ -127,43 +143,64 @@ const defaultFormState = {
   destination: '',
   startMileage: '',
   endMileage: '',
+  passengers: [],
 };
 
 export default {
   name: 'RouteFormView',
+  components: {
+    MultiSelect,
+  },
   data() {
     return {
       route: { ...defaultFormState },
       errors: {},
+      searchText: '',
+      selectedPassengers: [],
+      lastSelectPassenger: {},
     };
   },
   methods: {
     ...mapActions([actions.SUBMIT]),
     ...mapActions(namespaces.drives, [apiActions.fetchDrives]),
     ...mapActions(namespaces.cars, [apiActions.fetchCars]),
     ...mapActions(namespaces.passengers, [apiActions.fetchPassengers]),
+    onPassengerSelect(passengers, lastSelectPassenger) {
+      this.selectedPassengers = passengers;
+      this.lastSelectPassenger = lastSelectPassenger;
+      this.route.passengers = passengers.map(i => i.value);
+    },
     handleSubmit() {
       this.validateForm();
 
       if (!Object.keys(this.errors).length) {
         this[actions.SUBMIT]({ form: this.route });
         this.route = { ...defaultFormState };
+        this.selectedPassengers = [];
       }
     },
 
     validateForm() {
+      const makeErrorsPartial = makeErrors(this.$t.bind(this));
+
       const data =
         Object.entries(this.route).reduce((acc, [key, value]) =>
-          ({ ...acc, [key]: String(value).trim() }), {});
-
-      const makeErrorsPartial = makeErrors(this.$t.bind(this));
+          ({
+            ...acc,
+            [key]: stringFields.includes(key)
+              ? String(value).trim()
```

Comment:
String fields have string names, but form values could be numbers.

Reply:
What I meant is that all required fields are stringified, which is not desired for distances.

Reply:
Why is it not? Trim doesn't exist on numbers, where javascrtip evaluates strings as numbers so for eaxmple `'5' - '2' // 3' Also when it is sent to api, it will be sent as a string